fcrepo4-labs/fcrepo-import-export
# Export serialized bags

Resolves: https://jira.lyrasis.org/browse/FCREPO-3206

* Add command line argument for specifying serialization and format for export
  * --bag-serialization | -s
* Serialize bags when specified during export
* Update README to include documentation for bag serialization

------------------------------------------

## Testing

1. Start a fresh fedora repository, e.g. with fcrepo4-docker `FEDORA_TAG=5.1.0 docker-compose up -d`
2. Create a binary in the repository
3. Create a bag-config.yml for export
```
bag-info.txt:
  Source-Organization: fcrepo-import-export
  Organization-Address: localhost
  External-Description: Serialized BagIt Bag
```
4. Run the exporter
    * Can be done for each of `tar`, `tgz`, and `zip`
```
java -jar target/fcrepo-import-export-0.4.0-SNAPSHOT.jar -s tar --mode export --resource http://localhost:8080/fcrepo/rest --dir fcrepo-3206  --binaries --bag-profile beyondtherepository --bag-config bag-config.yml --user fedoraAdmin:secret3
```
5. Check that the exported bag is valid using either tar or unzip
    * `tar tvf fcrepo-3206.tar`
    * `tar tzvf fcrepo-3206.tar.gz`
    * `unzip -l fcrepo-3206 .zip`
